src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1288,50 +1288,60 @@ private Iterator<Resource> queryAllVanityPaths(String query) {
         return it;
     }
 
+    /**
+     * Utility class for running paged queries.
+     */
     private class PagedQueryIterator implements Iterator<Resource> {
```

Comment:
would it make sense to make it a full class now (instead of an inner class)?

Reply:
Maybe once we need it outside MapEntries. Wanted to miminize the diff for now.